ambry-store/src/test/java/com.github.ambry.store/StoreFindTokenTest.java

```diff
@@ -141,25 +146,17 @@ public void serDeTest() throws IOException {
   @Test
   public void constructionErrorCasesTest() {
     UUID sessionId = new UUID(0, 0);
+    UUID incarnationId = new UUID(0, 0);
     String logSegmentName = LogSegmentNameHelper.generateFirstSegmentName(numSegments);
     Offset offset = new Offset(logSegmentName, 0);
     MockId key = new MockId(UtilsTest.getRandomString(10));
 
-    // no offset in JournalBased
-    try {
-      new StoreFindToken(null, sessionId, null, false);
-      fail("Construction of StoreFindToken should have failed");
-    } catch (IllegalArgumentException e) {
-      // expected. Nothing to do.
-    }
-
-    // no sessionId in JournalBased
-    try {
-      new StoreFindToken(offset, null, null, false);
-      fail("Construction of StoreFindToken should have failed");
-    } catch (IllegalArgumentException e) {
-      // expected. Nothing to do.
-    }
+    // no offset
+    testConstructionFailure(key, sessionId, incarnationId, null);
+    // no session id
+    testConstructionFailure(key, null, incarnationId, offset);
+    // no incarnation Id
+    // testConstructionFailure(key, sessionId, null, offset);
```

Comment:
I couldnt' check for null incarnationId since we have to support reading of old versions as well. But will uncomment it once we completely move to the latest version